browsergym/experiments/src/browsergym/experiments/loop.py

```diff
@@ -657,6 +657,94 @@ def summary_info(self) -> dict:
                 self._summary_info = json.load(f)
         return self._summary_info
 
+    @property
+    def tape(self) -> dict:
+        """
+        TapeAgents (https://github.com/ServiceNow/TapeAgents) framework compatibility.
+        Exports experiment trace in the format of serialized tape.
+        Reuses tape segments if they were already placed in the agent_info during the experiment.
+
+        :returns: dict: serialized tape of the experiment
+        """
+        steps = []
+        for step_info in self.steps_info:
+            if "tape_segment" in step_info.agent_info["extra_info"]:
+                tape_segment = step_info.agent_info["extra_info"]["tape_segment"]
+            else:
+                tape_segment = self._create_tape_segment(step_info)
+            steps += tape_segment
+        metadata = dict(
+            id=str(uuid.uuid4()),
+            author="browsergym_agent",
+            result=self.get_exp_record(),
+        )
+        return dict(steps=steps, metadata=metadata)
+
+    def _create_tape_segment(self, step_info: StepInfo) -> list[dict]:
+        tape_segment = []
+        # extract observation step
+        if step_info.obs is not None:
+            screenshot: str = ""
+            screenshot_som: str = ""
+            obs_dict = step_info.obs.copy()
```

Comment:
Should this be a deepcopy?
Drawback: it's costly
Pro: tapes can safely be altered afterwards

Reply:
Yep, it's a good idea. I assumed earlier that the observation fields are mostly primitive types and will be copied anyway, but it's safer to make a deep copy. As the export should rarely be called, complexity is not an issue here. I'll update it, thanks!